### DISCUSSION TITLE
Failure to connect thru SSH

### DISCUSSION THREAD

**Discussion by oneeyeman1** [community/community#51083] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, ALL,
A while back Mr. Josh Soref helped me set up my machines with the SSH access n all 3 major platforms: Windows, Linux and OSX.

However, recently GitHub changed the key in their environment and I started getting error.

Googling revealed this: https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key and so I tried to follow the instructions there

Problem is: the very first instruction failed (as below):

[code]
Igors-MacBook-Air:dbhandler igorkorot$ ssh-keygen -R [github.com](http://github.com/)
Host [github.com](http://github.com/) not found in /Users/igorkorot/.ssh/known_hosts
Igors-MacBook-Air:dbhandler igorkorot$
[/code] 

Then I tried to continue and got following:

[code]
Igors-MacBook-Air:dbhandler igorkorot$ ssh-keygen -R [github.com](http://github.com/)
Host [github.com](http://github.com/) not found in /Users/igorkorot/.ssh/known_hosts
Igors-MacBook-Air:dbhandler igorkorot$ vi ~/.ssh/known_hosts
[/code]

At this point I figured I can simply edit the file manually and so I did.
But then it failed to connect with the same error.

Trying to investigate I saw an entry in the known_hosts file which was starting with:

[code]
[[ssh.github.com](http://ssh.github.com/)]:443,[140.82.113.36]:443
[/code]

so I commented it out and hoped it would work.

It did, however, everytime I try push/pull t asks me if I want to connect:

[code]
C:\Users\Igor\OneDrive\Documents\dbhandler_app [master ↑2 +8 ~2 -0 !]> git push
The authenticity of host '[[ssh.github.com](http://ssh.github.com/)]:443 ([140.82.114.35]:443)'
can't be established.
ECDSA key fingerprint is SHA256:p2QAMXNIC1TJYWeIOttrVc98/R1BUFWu3/LiyKgUfQM.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '[[ssh.github.com](http://ssh.github.com/)]:443' (ECDSA) to the list
of known hosts.
Warning: the ECDSA host key for '[[ssh.github.com](http://ssh.github.com/)]:443' differs from
the key for the IP address '[140.82.114.35]:443'
Offending key for IP in /c/Users/Igor/.ssh/known_hosts:6
Are you sure you want to continue connecting (yes/no)? yes
[/code]

I think there is something wrong with the instructions on the link I referenced..

Could someone please review them and fix them?

And if someone can explain how do I get rid of continuous question about whether or not I want to connect to GitHub, it would be great.

Thank you.

**Comment by oneeyeman1** [comment]
@lemur73 ,
This is different issue so you are probably better off with making a new thread..

Thank you

**Comment by airtower-luna** [comment]
1. Before accepting a new key for a GitHub SSH server (`github.com` or `ssh.github.com`), check if the fingerprint is one of those [published in the documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints) (the one for the key type you see). Otherwise someone might intercept your connection. :warning: 
2. You can delete conflicting keys for IP addresses the same way you can delete the old RSA keys, for example with the conflicting key mentioned in your log: `ssh-keygen -R '[140.82.114.35]:443'`

**Reply by oneeyeman1** [reply]
@airtower-luna ,
2. I know I can delete the offending line (the old key). But even if I do - I still get the same warning:

[quote]
Do you want to continue? (yes/no)
[/quote]

1. The problem is - the "official" GitHib documentation is misleading at best and at worse - is plain wrong.
Someone needs to go and fix it, because the keys that are present in the .ssh/known_hosts file are different and so they can't be deleted that way.

Thank you.
